include/exadg/grid/grid_motion_elasticity.h

```diff
@@ -56,16 +56,35 @@ class GridMotionElasticity : public GridMotionBase<dim, Number>
       pcout(std::cout,
             dealii::Utilities::MPI::this_mpi_process(
               structure_operator->get_dof_handler().get_communicator()) == 0),
-      iterations({0, {0, 0}})
+      iterations({0, {0, 0}}),
+      time_step_number_last_preconditioner_update(0)
   {
     pde_operator->initialize_dof_vector(displacement);
   }
 
+  /**
+   * Check if the preconditioner should be updated or was updated at this time step already
+   */
+  bool
+  update_preconditioner(types::time_step const time_step_number)
+  {
+    bool const update_preconditioner =
+      this->param.update_preconditioner &&
+      time_step_number % this->param.update_preconditioner_every_time_steps == 0 &&
+      time_step_number > time_step_number_last_preconditioner_update;
+
+    time_step_number_last_preconditioner_update = time_step_number;
+
+    return update_preconditioner;
```

Comment:
the update is only done once per time step (if the update is desired every _n_ time steps).